### PR TITLE
refactor staking/liquid amount conversion

### PR DESCRIPTION
**Pull request by GopherJ**
Signed-off-by: Cheng JIANG <alex_cj96@foxmail.com>

close: #1283

### CODE REVIEW

File: pallets/liquid-staking/src/lib.rs [hunk 1]
```diff
@@ -788,6 +771,27 @@ pub mod pallet {
...
+        fn do_update_exchange_rate(bonding_amount: BalanceOf<T>) -> DispatchResult {
+            let old_exchange_rate = Self::exchange_rate();
+            let matching_pool = Self::matching_pool();
+            match Rate::checked_from_rational(
+                bonding_amount
+                    .checked_add(matching_pool.total_stake_amount)
+                    .ok_or(ArithmeticError::Overflow)?,
+                T::Assets::total_issuance(Self::liquid_currency()?)
+                    .checked_add(matching_pool.total_unstake_amount)
+                    .ok_or(ArithmeticError::Overflow)?,
```
Comment (GopherJ) [inline pallets/liquid-staking/src/lib.rs:784]: @MrPai  will it cause issues if we use the edited matching_ledger to update exchange_rate? if yes I'll still need to pass old matching_pool to this function

Comment (MrPai) [reply pallets/liquid-staking/src/lib.rs:784]: We didn't think about what was matched

**Review by MrPai — COMMENTED** [review]
others looks good to me.

File: pallets/liquid-staking/src/lib.rs (hunk 1 above)
Comment (GopherJ) [reply pallets/liquid-staking/src/lib.rs:784]: sorry can you elaborate more?

Comment (GopherJ) [reply pallets/liquid-staking/src/lib.rs:784]: ok so you still want to use old matching_ledger right?

Comment (MrPai) [reply pallets/liquid-staking/src/lib.rs:786]: `old_matching_ledger` looks clear for now, however  edited matching_ledger maybe confused.

**Pull request merged**